packages/server/src/lib/grantsCollaboration/notes.js

```diff
@@ -147,9 +147,23 @@ async function getOrganizationNotesForGrant(
     return getCurrentNoteRevisions(knex, { grantId, organizationId }, { cursor, limit });
 }
 
+async function deleteGrantNotesByUser(knex, grantId, userId) {
+    const grantNote = knex
+        .select('id')
+        .from('grant_notes')
+        .where({ grant_id: grantId, user_id: userId });
+
+    await knex('grant_notes_revisions')
+        .whereIn('grant_note_id', grantNote)
+        .del();
+
+    await grantNote.del();
```

Comment:
I think it would be preferable to handle this on the Postgres side using `ON DELETE CASCADE`. Unfortunately, that doesn't currently seem to be implemented on the FK constraint, so it's probably best that we do that now since there's no data in Production just yet.

A few details that feel worth mentioning:
- Postgres doesn't support this kind of alteration on foreign keys, so the migration will need to drop the existing FK constraint and then re-add it with the `ON DELETE CASCADE` specification.
- Knex seems to have its own naming convention for foreign key constraints (it doesn't follow the Postgres standard convention of `{tablename}_{columnname(s)}_{suffix}`) so rather than using `knex.raw()`, it might be best to use the Knex schema builder functions (see [`table.dropForeign()`](https://knexjs.org/guide/schema-builder.html#dropforeign) and [`table.foreign()`](https://knexjs.org/guide/schema-builder.html#foreign)).

Reply:
yes certainly makes sense here - updated!